### PR TITLE
Improve python binding

### PR DESCRIPTION
**Pull request by tfiedor**
This PR introduces the following changes:

1. It does minor refactoring of the code (renaming of few functions, adding few helper functions
2. Add visualization of the NFA, including user-defined rules for coloring graphviz graphs
3. Add example jupyter notebooks
4. Renames mata to libmata for python binding to avoid clash with existing pypi mata
5. Add github action for automatic upload to pypi

### CODE REVIEW

**Comment by tfiedor** [comment]
Finally, I there are no dependencies I'm waiting for, so I request review of this crap, which is long due. I will introduce some fixes and improvements in near future, but in smaller batches. I will fix the fails as well, there will be some clashes in python dependencies which I forgot to add to requirements.

File: bindings/python/libmata.pyx [hunk 1]
```diff
@@ -1116,7 +1177,8 @@ cdef class Nfa:
 
     @classmethod
     def equivalence_check(cls, Nfa lhs, Nfa rhs, Alphabet alphabet = None, params = None) -> bool:
-        """Test equivalence of two automata.
+        """
+        Test equivalence of two automata.
```
Comment (Adda0) [inline bindings/python/libmata.pyx:1181]: Did we not agree on having the brief description already on the first line with `"""`?

File: bindings/python/tests/test_trans.py [hunk 2]
```diff
@@ -43,10 +43,10 @@ def test_transsymbstates():
 
 def test_transition_operations(prepare_automaton_a):
     nfa = mata.Nfa(10)
-    nfa.add_trans_raw(3, ord('c'), 4)
+    nfa.add_transition(3, ord('c'), 4)
     assert nfa.has_trans_raw(3, ord('c'), 4)
     trans = mata.Trans(4, ord('c'), 5)
-    nfa.add_trans(trans)
+    nfa.add_transition_object(trans)
```
Comment (Adda0) [inline bindings/python/tests/test_trans.py:49]: If I am not mistaken, we have decided in the meeting that we will remove this overload of former `add_trans()`, now `add_transition()` completely, so the following might be no longer relevant, but I am not necessarily happy with the use of `object`. That being said, I cannot come up with a better name anyway. We can keep `add_transition_object()`.

**Review by Adda0 — APPROVED** [review]
Having to version the outputs of Jupyter notebooks is pretty annoying, but we cannot do anything else in order to show the results to the user here on GH, unfortunately.

Other than that, this PR looks fine to me. The visualization is particularly great. Well done.

**Comment by tfiedor** [comment]
> Having to version the outputs of Jupyter notebooks is pretty annoying, but we cannot do anything else in order to show the results to the user here on GH, unfortunately.
> 
> Other than that, this PR looks fine to me. The visualization is particularly great. Well done.

Yes, I agree, however, it is nice to see some outputs in the github. Other than that the notebooks should be in `.gitignore` to prevent accidental pushes (hopefully).

File: bindings/python/tests/test_trans.py (hunk 2 above)
Comment (tfiedor) [reply bindings/python/tests/test_trans.py:49]: I have no issue with removing this function. Will do it tommorow. I'm satisfied with the single one.

File: bindings/python/libmata.pyx (hunk 1 above)
Comment (tfiedor) [reply bindings/python/libmata.pyx:1181]: Yes, probably type. Thanks, will fix this :-)

**Comment by Adda0** [comment]
Exactly. Sounds good to me.

**Review by martinhruska — APPROVED** [review]

**Comment by martinhruska** [comment]
I have no issues with PR but I would consider moving jupyter notebook examples to a special directory. Either to a folder `exmples/notebooks` or a folder `notebooks`. I would not mess .cc exmaples with jupyter ones since they would both interest different people.

**Comment by tfiedor** [comment]
> I have no issues with PR but I would consider moving jupyter notebook examples to a special directory. Either to a folder `exmples/notebooks` or a folder `notebooks`. I would not mess .cc exmaples with jupyter ones since they would both interest different people.

OK, sounds fine to me. I will fix these stuff later today and then merge the shit out of this myself.

File: bindings/python/tests/test_trans.py (hunk 2 above)
Comment (tfiedor) [reply bindings/python/tests/test_trans.py:49]: I rethough this, and will fix it after Martin's cleanup. I skimmed through the bindings, and there are more of these overloaded versions, that should be removed.

**Pull request merged**